app/bundles/LeadBundle/Model/CompanyModel.php

```diff
@@ -516,6 +516,69 @@ public function removeLeadFromCompany($companies, $lead): void
         unset($lead, $deleteCompany, $companies);
     }
 
+    /**
+     * Get a list of companies with names only.
+     *
+     * @param string         $type
+     * @param mixed[]|string $filter
+     * @param int            $limit
+     * @param int            $start
+     * @param string         $exclude
+     * 
+     * @return array[]
+     */
+    public function getSimpleLookupResults($type, $filter = '', $limit = 10, $start = 0, $exclude = ''): array
+    {
+        $results = [];
+        $valueColumn = 'id';
+        $onlyNames = true;
+        switch ($type) {
+            case 'companyfield':
+            case 'lead.company':
```

Comment:
The Cyclomatic complexity could be reduced by returning early:
```suggestion
        if (!in_array($type, ['companyfield', 'lead.company']) {
            return [];
        }
```
Then the rest of the code can move 2 tabs to the left.